lib/vrf.c

```diff
@@ -424,32 +446,86 @@ int vrf_socket(int domain, int type, int protocol, vrf_id_t vrf_id)
 	return ret;
 }
 
-/* vrf CLI commands */
-DEFUN_NOSH (vrf,
-       vrf_cmd,
-       "vrf NAME",
-       "Select a VRF to configure\n"
-       "VRF's name\n")
+extern int vrf_is_backend_ns(void)
+{
+	return vrf_backend_netns;
+}
+
+void vrf_configure_backend(int ns_on)
+{
+	vrf_backend_netns = ns_on;
+}
+
+int vrf_handler_create(struct vty *vty, const char *vrfname, struct vrf **vrf)
 {
-	int idx_name = 1;
-	const char *vrfname = argv[idx_name]->arg;
 	struct vrf *vrfp;
 
 	if (strlen(vrfname) > VRF_NAMSIZ) {
-		vty_out(vty,
-			"%% VRF name %s is invalid: length exceeds "
-			"%d characters\n",
-			vrfname, VRF_NAMSIZ);
+		if (vty)
+			vty_out(vty,
+			    "%% VRF name %s is invalid: length exceeds "
+			    "%d characters\n", vrfname, VRF_NAMSIZ);
+		else
+			zlog_warn("%% VRF name %s is invalid: length exceeds "
+				  "%d characters\n", vrfname, VRF_NAMSIZ);
 		return CMD_WARNING_CONFIG_FAILED;
 	}
 
 	vrfp = vrf_get(VRF_UNKNOWN, vrfname);
 
-	VTY_PUSH_CONTEXT(VRF_NODE, vrfp);
+	if (vty)
+		VTY_PUSH_CONTEXT(VRF_NODE, vrfp);
 
+	if (vrf)
+		*vrf = vrfp;
 	return CMD_SUCCESS;
 }
 
+int vrf_is_mapped_on_netns(struct vrf *vrf)
+{
+	if (!vrf || vrf->data.ns_name[0] == '\0')
+		return 0;
+	if (vrf->vrf_id == NS_DEFAULT)
+		return 0;
+	return 1;
+}
+
+int vrf_switch_to_netns(struct vrf *vrf)
+{
+	char *name;
+
+	if (!vrf || vrf->data.ns_name[0] == '\0')
+		return -1;
+	if (vrf->vrf_id == NS_DEFAULT)
+		return -1;
+	name = ns_netns_pathname(NULL, vrf->data.ns_name);
+	if (debug_vrf)
+		zlog_debug("VRF_SWITCH: %s(%u)", name, vrf->vrf_id);
+	return ns_switch_to_netns (name);
+}
+
+int vrf_switchback_to_initial (void)
```

Comment:
Again this function is never used and equivalent ns version?

Reply:
same as before:

- vrf_switchback_to_initial() is the wrapper for ns_switchback_to_initial().